remoting/subsystem/src/main/java/org/jboss/as/remoting/RemotingSubsystemRootResource.java

```diff
@@ -106,65 +102,30 @@ public class RemotingSubsystemRootResource extends SimpleResourceDefinition {
-        }
-    }
-
-    //The defaults for these come from XnioWorker
-    static final SimpleAttributeDefinition WORKER_READ_THREADS = createWorkerAttribute(CommonAttributes.WORKER_READ_THREADS, Attribute.WORKER_READ_THREADS, 1);
-    static final SimpleAttributeDefinition WORKER_TASK_CORE_THREADS = createWorkerAttribute(CommonAttributes.WORKER_TASK_CORE_THREADS, Attribute.WORKER_TASK_CORE_THREADS, 4);
-    static final SimpleAttributeDefinition WORKER_TASK_KEEPALIVE = createWorkerAttribute(CommonAttributes.WORKER_TASK_KEEPALIVE, Attribute.WORKER_TASK_KEEPALIVE, 60);
-    static final SimpleAttributeDefinition WORKER_TASK_LIMIT = createWorkerAttribute(CommonAttributes.WORKER_TASK_LIMIT, Attribute.WORKER_TASK_LIMIT, 0x4000);
-    static final SimpleAttributeDefinition WORKER_TASK_MAX_THREADS = createWorkerAttribute(CommonAttributes.WORKER_TASK_MAX_THREADS, Attribute.WORKER_TASK_MAX_THREADS, 16);
-    static final SimpleAttributeDefinition WORKER_WRITE_THREADS = createWorkerAttribute(CommonAttributes.WORKER_WRITE_THREADS, Attribute.WORKER_WRITE_THREADS, 1);
-
-
-    static SimpleAttributeDefinition[] LEGACY_ATTRIBUTES = new SimpleAttributeDefinition[]{
-            WORKER_READ_THREADS,
-            WORKER_TASK_CORE_THREADS,
-            WORKER_TASK_KEEPALIVE,
-            WORKER_TASK_LIMIT,
-            WORKER_TASK_MAX_THREADS,
-            WORKER_WRITE_THREADS
-    };
+    static final PathElement PATH = PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, RemotingExtension.SUBSYSTEM_NAME);
 
-    private static final PathElement PATH = PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, RemotingExtension.SUBSYSTEM_NAME);
+    private final AttributeDefinition workerAttribute;
 
-    static ResourceDefinition create(ProcessType processType, boolean forDomain) {
-        return new RemotingSubsystemRootResource(new Attributes(processType, forDomain));
-    }
-
-    private final Attributes attributes;
-
-    private RemotingSubsystemRootResource(Attributes attributes) {
+    RemotingSubsystemRootResource(ProcessType processType) {
         super(new Parameters(PATH, RemotingExtension.getResourceDescriptionResolver(RemotingExtension.SUBSYSTEM_NAME))
-                .setAddHandler(new RemotingSubsystemAdd(attributes))
+                .setAddHandler(new RemotingSubsystemAdd())
                 .setAddRestartLevel(OperationEntry.Flag.RESTART_NONE)
                 .setRemoveHandler(new ReloadRequiredRemoveStepHandler())
                 .setRemoveRestartLevel(OperationEntry.Flag.RESTART_ALL_SERVICES)
                 .setCapabilities(REMOTING_ENDPOINT_CAPABILITY, HTTP_LISTENER_REGISTRY_CAPABILITY)
         );
-        this.attributes = attributes;
+        // Worker is not required if endpoint is meant to be supplied by DomainServerCommunicationServices
+        this.workerAttribute = new SimpleAttributeDefinitionBuilder(WORKER).setRequired(processType != ProcessType.DOMAIN_SERVER).build();
```

Comment:
I think that means it's not required on an HC either; i.e. domain.xml doesn't need to specify it if the server doesn't need it.

Reply:
This logic was based on https://github.com/wildfly/wildfly-core/blob/main/remoting/subsystem/src/main/java/org/jboss/as/remoting/RemotingSubsystemAdd.java#L83
https://github.com/wildfly/wildfly-core/blob/main/remoting/subsystem/src/main/java/org/jboss/as/remoting/RemotingSubsystemAdd.java#L83

But, to your point, ProcessType.HOST_CONTROLLER would not execute any runtime phase, so the model should not otherwise require this, I think.

Reply:
Replaced conditional with ProcessType.isManagedDomain(), which should cover all use cases, I think.